src/main/java/software/amazon/awssdk/crt/io/TlsKeyOperation.java

```diff
@@ -0,0 +1,189 @@
+        return inputData;
+    }
+
+    /**
+     * Returns the operation that needs to be performed.
+     *
+     * @return The operation that needs to be performed.
+     */
+    public Type getType() {
+        return operationType;
+    }
+
+    /**
+     * Returns the TLS algorithm used in the signature.
+     *
+     * @return The TLS algorithm used in the signature
+     */
+    public TlsSignatureAlgorithm getSignatureAlgorithm() {
+        return signatureAlgorithm;
+    }
+
+    /**
+     * Returns the TLS Hash algorithm used in the digest.
+     *
+     * @return The TLS Hash algorithm used in the digest
+     */
+    public TlsHashAlgorithm getDigestAlgorithm() {
+        return digestAlgorithm;
+    }
+
+    /**
+     * The function to call when you have modified the input data using the private key and are ready to
+     * return it for use in the mutual TLS Handshake.
+     *
+     * @param output The modified input data that has been modified by the custom key operation
+     */
+    public synchronized void complete(byte[] output) {
+        if (nativeResource.internalGetNativeHandle() == 0) {
+            Log.log(LogLevel.Error, LogSubject.CommonGeneral,
+                "No native handle set in TlsKeyOperation! Cannot complete operation");
+            return;
+        }
+
+        tlsKeyOperationComplete(nativeResource.internalGetNativeHandle(), output);
+        nativeResource.close();
+    }
+
+    /**
+     * The function to call when you either have an exception and want to complete the operation with an
+     * exception or you cannot complete the operation. This will mark the operation as complete with an
+     * exception so it can be reacted to accordingly.
+     *
+     * @param ex The exeception to complete with
+     */
+    public synchronized void completeExceptionally(Throwable ex) {
+        if (nativeResource.internalGetNativeHandle() == 0) {
+            Log.log(LogLevel.Error, LogSubject.CommonGeneral,
+                "No native handle set in TlsKeyOperation! Cannot complete operation exceptionally");
+            return;
+        }
+
+        tlsKeyOperationCompleteExceptionally(nativeResource.internalGetNativeHandle(), ex);
+        nativeResource.close();
+    }
+
+    /**
+     * The TlsKeyOperation has special lifetime rules, where you have to call one of the complete functions, and
+     * by using this private, internal-only CRT resource, we can still get the benefits of using a CRT resource
+     * for detecting memory leaks, while not exposing functionality that would conflict with the lifetime rules.
+     */
+    private class CrtResourceInternal extends CrtResource {
+        protected void releaseNativeHandle() {}
+
+        protected boolean canReleaseReferencesImmediately() {
+            return true;
+        };
+
+        /* Exposing acquireNativeHandle */
+        public void internalAcquireNativeHandle(long handle) {
+            acquireNativeHandle(handle);
+        }
```

Comment:
trivial: Just have your constructor take the `long handle` and then the constructor can call the protected functions